File: DropDownList/CustomItemWithMultipleImages/RadListControlCustomItemsVB/Helpers/CustomListDataItem.vb
Imports Telerik.WinControls.UI
Imports Telerik.WinControls
Imports System.ComponentModel

''' <summary>
''' The custom list data item.
''' </summary>
Public Class CustomListDataItem
    Inherits RadListDataItem
#Region "Constants and Fields"

    ''' <summary>
    ''' The image property.
    ''' </summary>
    Public Shared ReadOnly ImageProperty As RadProperty = RadProperty.Register("Image", GetType(Image), GetType(CustomListDataItem), New RadElementPropertyMetadata(Nothing))

    ''' <summary>
    ''' The image 2 property.
    ''' </summary>
    Public Shared ReadOnly Image2Property As RadProperty = RadProperty.Register("Image2", GetType(Image), GetType(CustomListDataItem), New RadElementPropertyMetadata(Nothing))

    ''' <summary>
    ''' The name property.
    ''' </summary>
    Public Shared ReadOnly NameProperty As RadProperty = RadProperty.Register("Name", GetType(String), GetType(CustomListDataItem), New RadElementPropertyMetadata(String.Empty))

#End Region

#Region "Properties"

    ''' <summary>
    ''' Gets or sets Image.
    ''' </summary>
    Public Shadows Property Image() As Image
        Get
            Return DirectCast(Me.GetValue(ImageProperty), Image)
        End Get

        Set(ByVal value As Image)
            Me.SetValue(ImageProperty, value)
        End Set
    End Property

    ''' <summary>
    ''' Gets or sets Image2.
    ''' </summary>
    Public Property Image2() As Image
        Get
            Return DirectCast(Me.GetValue(Image2Property), Image)
        End Get

        Set(ByVal value As Image)
            Me.SetValue(Image2Property, value)
        End Set
    End Property

    ''' <summary>
    ''' Gets or sets Name.
    ''' </summary>
    Public Property Name() As String
        Get
            Return DirectCast(Me.GetValue(NameProperty), String)
        End Get

        Set(ByVal value As String)
            Me.SetValue(NameProperty, value)
        End Set
    End Property

#End Region

#Region "Methods"

    ''' <summary>
    ''' The set data bound item.
    ''' </summary>
    ''' <param name="dataBinding">
    ''' The data binding.
    ''' </param>
    ''' <param name="value">
    ''' The value.
    ''' </param>
    Protected Overrides Sub SetDataBoundItem(ByVal dataBinding As Boolean, ByVal value As Object)
        MyBase.SetDataBoundItem(dataBinding, value)
        If TypeOf value Is INotifyPropertyChanged Then
            Dim item = TryCast(value, INotifyPropertyChanged)
            AddHandler item.PropertyChanged, AddressOf item_PropertyChanged
        End If
    End Sub

    ''' <summary>
    ''' The item_ property changed.
    ''' </summary>
    ''' <param name="sender">
    ''' The sender.
    ''' </param>
    ''' <param name="e">
    ''' The e.
    ''' </param>
    Private Sub item_PropertyChanged(ByVal sender As Object, ByVal e As PropertyChangedEventArgs)
        If e.PropertyName = "Image" Then
            Me.Image = TryCast(Me.DataBoundItem, BussinessObject).Image
        End If

        If e.PropertyName = "Image2" Then
            Me.Image2 = TryCast(Me.DataBoundItem, BussinessObject).Image2
        End If

        If e.PropertyName = "Name" Then
            Me.Name = TryCast(Me.DataBoundItem, BussinessObject).Name
        End If
    End Sub

#End Region
End Class

File: DropDownList/CustomItemWithMultipleImages/RadListControlCustomItemsCS/Helpers/CustomListDataItem.cs
using System.ComponentModel;
using System.Drawing;
using Telerik.WinControls;
using Telerik.WinControls.UI;

namespace RadListControlCustomItems.Helpers
{
    /// <summary>
    /// The custom list data item.
    /// </summary>
    public class CustomListDataItem : RadListDataItem
    {
        #region Constants and Fields

        /// <summary>
        /// The image 2 property.
        /// </summary>
        public static readonly RadProperty Image2Property = RadProperty.Register(
            "Image2", typeof(Image), typeof(CustomListDataItem), new RadElementPropertyMetadata(null));

        /// <summary>
        /// The image property.
        /// </summary>
        public static readonly RadProperty ImageProperty = RadProperty.Register(
            "Image", typeof(Image), typeof(CustomListDataItem), new RadElementPropertyMetadata(null));

        /// <summary>
        /// The name property.
        /// </summary>
        public static readonly RadProperty NameProperty = RadProperty.Register(
            "Name", typeof(string), typeof(CustomListDataItem), new RadElementPropertyMetadata(string.Empty));

        #endregion Constants and Fields

        #region Properties

        /// <summary>
        /// Gets or sets Image.
        /// </summary>
        public new Image Image
        {
            get
            {
                return (Image)this.GetValue(ImageProperty);
            }

            set
            {
                this.SetValue(ImageProperty, value);
            }
        }

        /// <summary>
        /// Gets or sets Image2.
        /// </summary>
        public Image Image2
        {
            get
            {
                return (Image)this.GetValue(Image2Property);
            }

            set
            {
                this.SetValue(Image2Property, value);
            }
        }

        /// <summary>
        /// Gets or sets Name.
        /// </summary>
        public string Name
        {
            get
            {
                return (string)this.GetValue(NameProperty);
            }

            set
            {
                this.SetValue(NameProperty, value);
            }
        }

        #endregion Properties

        #region Methods

        /// <summary>
        /// The set data bound item.
        /// </summary>
        /// <param name="dataBinding">
        /// The data binding.
        /// </param>
        /// <param name="value">
        /// The value.
        /// </param>
        protected override void SetDataBoundItem(bool dataBinding, object value)
        {
            base.SetDataBoundItem(dataBinding, value);
            if (value is INotifyPropertyChanged)
            {
                var item = value as INotifyPropertyChanged;
                item.PropertyChanged += this.item_PropertyChanged;
            }
        }

        /// <summary>
        /// The item_ property changed.
        /// </summary>
        /// <param name="sender">
        /// The sender.
        /// </param>
        /// <param name="e">
        /// The e.
        /// </param>
        private void item_PropertyChanged(object sender, PropertyChangedEventArgs e)
        {
            if (e.PropertyName == "Image")
            {
                this.Image = (this.DataBoundItem as BussinessObject).Image;
            }

            if (e.PropertyName == "Image2")
            {
                this.Image2 = (this.DataBoundItem as BussinessObject).Image2;
            }

            if (e.PropertyName == "Name")
            {
                this.Name = (this.DataBoundItem as BussinessObject).Name;
            }
        }

        #endregion Methods
    }
}